apps/frontend/src/libs/hooks/use-handle-click-outside/use-handle-click-outside.hook.ts

```diff
@@ -3,9 +3,14 @@ import { useEffect } from "~/libs/hooks/hooks.js";
 const useHandleClickOutside = <T extends HTMLElement>(
 	reference: React.RefObject<T>,
 	onOutsideClick: () => void,
+	contentReference?: React.RefObject<T>,
 ): void => {
 	useEffect(() => {
 		const handleClickOutside = (event: MouseEvent): void => {
+			if (contentReference && !contentReference.current) {
+				return;
+			}
```

Comment:
What is this contentReference prop for? Looks like it'll only stop function if there is no content ref passed. If so, then I don't think there is no logic we want to have here.

Reply:
While I was working with notification popover, I noticed that when popover is closed, and you click anywhere on the page, onPopoverClose function was always running. With these changes it only runs after the popover was open and you click outside this area.

Reply:
Oh, strange. It's good that you've noticed this, thanks for explanation!